src/v2/manifest/mod.rs

```diff
@@ -234,3 +260,100 @@ fn to_mimes(v: &[&str]) -> Result<Vec<mime::Mime>> {
+        }
+        (None, false) => Err(Error::from(
+            "no header_content_type given and no workaround to apply".to_string(),
+        )),
+        (Some(header_value), true) => {
+            // TODO: remove this workaround once Satellite returns a proper content-type here
+            match header_value {
+                "application/x-troff-man" => {
+                    trace!("Applying workaround for pulp-based registries, e.g. Satellite");
+                    mediatypes::MediaTypes::from_str(
+                        "application/vnd.docker.distribution.manifest.v1+prettyjws",
+                    )
+                    .map_err(Into::into)
+                }
+                _ => {
+                    debug!("Received content-type '{}' from pulp-based registry. Feeling lucky and trying to parse it...", header_value);
+                    mediatypes::MediaTypes::from_str(header_value).map_err(Into::into)
+                }
+            }
+        }
+        (None, true) => {
+            trace!("Applying workaround for pulp-based registries, e.g. Satellite");
+            mediatypes::MediaTypes::from_str(
+                "application/vnd.docker.distribution.manifest.v1+prettyjws",
+            )
+            .map_err(Into::into)
+        }
+    }
+}
+
+/// Umbrella type for common actions on the different manifest schema types
+#[derive(Debug)]
+pub enum Manifest {
+    S1Signed(manifest_schema1::ManifestSchema1Signed),
+    S2(manifest_schema2::ManifestSchema2),
+    ML(manifest_schema2::ManifestList),
+}
+
+impl Manifest {
+    /// List digests of all layers referenced by this manifest, if available.
+    ///
+    /// The returned layers list is ordered starting with the base image first.
+    pub fn layers_digests(&self) -> Result<Vec<String>> {
+        match self {
+            Manifest::S1Signed(m) => Ok(m.get_layers()),
+            Manifest::S2(m) => Ok(m.get_layers()),
+            // Manifest::ML(_) => TODO(steveeJ),
+            _ => Err(format!(
+                "Manifest {:?} doesn't support the 'layer_digests' method",
+                self
+            )
+            .into()),
+        }
+    }
+
+    /// The architecture of the image the manifest points to, if available.
+    ///
+    /// For some Manifest types this could result in an external request, hence
+    /// the result is wrapped in an asynchronous type.
+    pub fn architecture(&self) -> Result<String> {
+        match self {
+            Manifest::S1Signed(m) => Ok(m.architecture.clone()),
+            Manifest::S2(m) => Ok(m.architecture()),
+            // Manifest::ML(_) => TODO(steveeJ),
+            _ => Err(format!(
+                "Manifest {:?} doesn't support the 'architecture' method",
+                self
+            )
+            .into()),
+        }
+    }
+
+    /// Get a collection of all image labels for the given layer, if available.
+    pub fn labels(&self, layer: usize) -> Result<Option<(HashMap<String, String>)>> {
```

Comment:
I have the feeling that this method does not belong here. Where are the labels for Schema2 and ManifestList types?

Reply:
I had the same feeling. I'm fine dropping it, since I'm also not aware of any consumers.

Reply:
Let's drop it then (assuming Cincinnati is not using it).